File: Core/Entities/Reservation.cs
namespace Core.Entities;

public class Reservation : BaseEntity
{
    public required string SurName { get; set; }
    public required string Name { get; set; }
    public required string Email { get; set; }
    public DateTime ReservationDate { get; set; }
    public int NumberOfAdults { get; set; }
    public int NumberOfChildren { get; set; }
    public decimal TotalPrice { get; set; }
    public string? Remark { get; set; }
    public bool IsPaid { get; set; }
    public string? PaymentRemark { get; set; }
    public required int ShowId { get; set; }
    public string? PaymentCode => $"Ortier-{Id:D3}";
}

File: client/src/app/shared/Models/Reservation.ts
export type Reservation = {
    id: number;
    email: string;
    showId: number;
    numberOfAdults: number;
    numberOfChildren: number;
    paymentCode: string | null;
    isPaid: boolean;
    surName: string;
    name: string;
    remark: string;
    reservationDate: string;
    totalPrice: number;
}